File: SharePreferences/app/src/main/java/com/example/sharepreferences/MainActivity.java
package com.example.sharepreferences;

import static com.example.sharepreferences.R.id.textView2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SharedPreferences sharedPreferences = this.getSharedPreferences("mySharedPreferences",
                Context.MODE_PRIVATE);

        SharedPreferences.Editor editor = sharedPreferences.edit();
//        editor.putString("newHelloText", "Hello Shared Preferences");
//        editor.remove("newHelloText");
//        editor.clear();
        editor.apply();

        TextView textView = findViewById(R.id.textView);
        textView.setText(sharedPreferences.getString("newHelloText", "Default text"));



    }
}

File: LayoutWorkout/app/src/main/java/com/example/layoutworkout/MainActivity.kt
package com.example.layoutworkout

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.databinding.DataBindingUtil
import com.example.layoutworkout.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var activityMainBinding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
       activityMainBinding = DataBindingUtil.setContentView(this,
           R.layout.activity_main)

        val myButton: Button = findViewById(R.id.button)
        myButton.setOnClickListener { changeText() }
    }

    private fun changeText() {
        val myTexView: TextView = findViewById(R.id.titletextView)

//        myTexView.text = "Data Binding"
        activityMainBinding.titletextView.text = "Data Binding"
    }
}